src/pilot/assign_role.py

```diff
@@ -1161,6 +1166,23 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
                     break
 
             if raid_size_gb == 0:
+                try:
+                    physical_disk_view_doc = drac_client.enumerate(
+                        DCIM_PhysicalDiskView)
+                    physical_disk_docs = utils.find_xml(
+                        physical_disk_view_doc,
+                        'DCIM_PhysicalDiskView',
+                        DCIM_PhysicalDiskView,
+                        True)
+                    disks = [get_size_in_bytes(physical_disk_doc,
+                                               DCIM_PhysicalDiskView)
+                             for physical_disk_doc in physical_disk_docs]
+                    os_volume_size_gb = int(disks[0]) / units.Gi
```

Comment:
Can we check to see if there is more than 1 disk, and raise an exception if so?

Reply:
I think it would be nice if we can only update the os_volume_size_gb for single disk. and add a note starting of block like "Below code block represents single disk scenario"

if len(disks) == 1:
     os_volume_size_gb = int(disks[0]) / units.Gi